crypto/s2n_rsa.c

```diff
@@ -120,6 +121,11 @@ static int s2n_rsa_encrypt(const struct s2n_pkey *pub, struct s2n_blob *in, stru
     /* Safety: RSA_public_encrypt does not mutate the key */
     int r = RSA_public_encrypt(in->size, (unsigned char *) in->data, (unsigned char *) out->data,
             s2n_unsafe_rsa_get_non_const(pub_key), RSA_PKCS1_PADDING);
+
+    if (s2n_is_in_fips_mode() && !s2n_libcrypto_is_awslc()) {
+        POSIX_ENSURE((int64_t) r != -1, S2N_ERR_INVALID_RSA_KEY_SIZE_IN_OSSL_FIPS_MODE);
+    }
+
     POSIX_ENSURE((int64_t) r == (int64_t) out->size, S2N_ERR_SIZE_MISMATCH);
```

Comment:
This is overkill and way too specific for such a niche failure case. You also can't be sure that the failure is due to the RSA key size, even with openssl-1.0.2. Encryption could be failing for a different reason.

You shouldn't check for the specific error. Just check for any error.
```suggestion
    int r = RSA_public_encrypt(in->size, (unsigned char *) in->data, (unsigned char *) out->data,
            s2n_unsafe_rsa_get_non_const(pub_key), RSA_PKCS1_PADDING);
    POSIX_ENSURE(r > 0, S2N_ERR_ENCRYPT);
    POSIX_ENSURE((int64_t) r == (int64_t) out->size, S2N_ERR_SIZE_MISMATCH);
```
You should also probably make the same change to [s2n_rsa_decrypt](https://github.com/aws/s2n-tls/blob/d88e8c8df2de472bf7a0f9ec94e33b21e1c73db5/crypto/s2n_rsa.c#L151) while you're at it.